dymos/phase/phase.py

```diff
@@ -1184,7 +1184,7 @@ def add_boundary_constraint(self, name, loc, constraint_name=None, units=None,
         bc_dict[name]['indices'] = indices
         bc_dict[name]['lower'] = lower
         bc_dict[name]['upper'] = upper
-        bc_dict[name]['equals'] = equals
+        bc_dict[name]['equals'] = list(equals) if type(equals) is np.ndarray else equals
```

Comment:
This isn't the problem. The real problem is in transcription_base.py when checking compatibility of the bounds. The check needs to be modified to take into account that these can be ndarrays as well as list or scaler.

dymos should support numpy arrays for 'equals', 'lower', and 'upper', and should pass those through to OpenMDAO, which supports numpy as well.